### PR TITLE
Exclude API docs from basic authentication requirement

### LINKED ISSUE #5353: Fix direct links to Generated API Docs page
Support opening the API docs link in a new browser (i.e. incognito mode).

Currently, opening the CiviForm home page in incognito, unauthenticated, followed by opening the API docs page (without logging in as an admin) works, but opening the API docs page in incognito mode directly causes you to get a sign-in prompt.

### PR DESCRIPTION
**Pull request by yotommy**
### Description

Exclude API docs from basic authentication requirement.

## Release notes

Exclude API docs from basic authentication requirement. They are meant to be [publicly visible](https://sourcegraph.com/github.com/civiform/civiform@9358f1b66ff84c74f953246520cf86544568dcb9/-/blob/server/app/views/api/ApiDocsView.java?L280-288).

#### General

Read the full guidelines for PRs [here](https://docs.civiform.us/contributor-guide/developer-guide/technical-contribution-guide#guidelines-for-pull-requests)

- [x] Added the correct label: < feature | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Created unit and/or browser tests which fail without the change (if possible)
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [ ] Extended the README / documentation, if necessary

### Issue(s) this completes

Fixes #5353

### CODE REVIEW

**Review by bion — COMMENTED** [review]
@yotommy possible to test this with a browser tes?

**Comment by avaleske** [comment]
+1 on browser tests, everything else is lgtm

**Comment by yotommy** [comment]
Added a browser test, which was an outstanding idea, as it exposed a bug: only admins could change programs on the API docs. Added `api-docs.ts` to the `applicant-bundle` to fix this.

The browser test encountered authentication errors (returning 401 even though it was using `AnonymousClient`), so I modified the existing regex to exclude the API docs path rather than defining a new pattern/client stanza.

File: server/app/assets/javascripts/applicant_entry_point.ts [hunk 1]
```diff
@@ -25,4 +26,5 @@ window.addEventListener('load', () => {
   azureDelete.init()
   azureUpload.init()
   phoneNumber.init()
+  apiDocs.init()
```
Comment (avaleske) [inline server/app/assets/javascripts/applicant_entry_point.ts:29]: suggestion (not blocking): would it be helpful to add a comment here explaining why the api docs are in the applicant bundle?

**Review by avaleske — APPROVED** [review]
Thank you!

**Review by bion — APPROVED** [review]
nice!

**Pull request merged**